opencti-platform/opencti-graphql/src/database/middleware.js

```diff
@@ -3405,7 +3405,8 @@ export const internalDeleteElementById = async (context, user, id, opts = {}) =>
       // Start by deleting external files
       const isTrashableElement = !isInferredIndex(element._index)
                 && (isStixCoreObject(element.entity_type) || isStixCoreRelationship(element.entity_type) || isStixSightingRelationship(element.entity_type));
-      const forceDelete = !!opts.forceDelete || !isTrashableElement;
+      // isModuleActivated from domain/settings cannot be used because of dependency cycle issues, we use the flag from conf instead
+      const forceDelete = !!opts.forceDelete || !isTrashableElement || !ENABLED_GARBAGE_COLLECTION_MANAGER;
```

Comment:
this part is not right, ENABLED_GARBAGE_COLLECTION_MANAGER can be false in a node but true in another (cluster mode). 
And we might need to validate this condition also (are we sure we want to delete permanently when garbage collector is disabled?)

Reply:
I agree, this is actually a functional question.